### DISCUSSION TITLE
Apple IIe Revision A Double-LoRes, & Double-HiRes

### DISCUSSION THREAD

**Discussion by pgreenland** [markadev/AppleII-VGA#43] [Q&A] [answered]
Hi All,

I've been looking at purchasing an Apple IIe to add to my collection of computing toys. It's a UK (so PAL) spec with a Rev A motherboard.

I understand that the Rev A's lacked double low and double high res support, which was added with the Rev B motherboard.

Any idea if the AppleII-VGA would allow for double res support on a Rev A board?

Looking at the revision history listed in "Understanding the Apple IIe" https://archive.org/details/Understanding_the_Apple_IIe/page/n353/mode/2up It appears, from a cursory glance and rough understanding, that many of the changes were in the video signal generation side of things. I was hoping that while the internal video hardware may be generating a garbage output, AppleII-VGA may be able to do its thing.

I'm only just beginning my Apple IIe journey (or considering at least) so wasn't sure how much of a limitation the lack of double res modes would be. I've found references to at least a few games that use them.

Any info would be greatly appreciated + what an amazing project!

Regards,

Phil

**Comment by markadev** [comment, marked as answer]
Hey there, that's a good question!

Rev A's and PAL units are pretty rare where I am so I couldn't say for sure, but I _think_ it would work like you described.  According to the chapter from "Understanding the Apple IIe" I didn't see any other documented changes to memory banking or soft-switches that would concern me.

The VGA card works by observing the memory accesses from the CPU and then simulating the entire video output subsystem so the video output circuitry on the main board could even be (almost) entirely non-functional and the VGA card would still work.

Congrats to you if you can get a Rev A machine!  I'll say that, in my opinion, the double-res modes are a big part of the IIe experience but it will really depend on the software you choose to run.

Also, unrelated to the VGA card compatibility, but since you're less familiar with Apple IIe's then there's another variation of the machine you should know about that can affect software compatibility.  Later in the IIe lifecycle Apple released an [Enhanced version](https://en.wikipedia.org/wiki/Apple_IIe#Enhanced_IIe).  Software that uses the newer CPU instructions would not run on the unenhanced models but older models could be upgraded by just replacing a few chips.

Best!

**Reply by pgreenland** [reply]
Hey Mark,

My offer was accepted....if all goes well I should be able to confirm fairly soon / I'll be scrabbling to try to find a display with a composite input. Using a modern (in comparison) VGA display would definitely be easier.

Thanks for the tips on the IIe vs the enhanced IIe. I'd seen a few references to the enhanced version, with the slightly later CPU. Given that everything is socketed in the Rev A, it might make for a nice easily reversible tweak sometime if I'm feeling brave.

Will mark your response as answered for now, as it might be a while before I can confirm either way. Will update the thread with how I get on.

Thanks for taking the time to look into it / reply!

Regards,

Phil
